### DISCUSSION TITLE
Downloader Plugin: How to write album covers & metadata to files downloaded at "Source"  quality?

### DISCUSSION THREAD

**Discussion by patperek** [pear-devs/pear-desktop#1939] [Q&A]
Hello, I've noticed that currently the downloader plugin will write album covers to downloaded mp3 files (mp3 256kbps), but if you do any other file format including "Source" file format (m4a), no album covers are applied to the files.

Is there any solution/config option to enable this? I've looked around on relevant discussions but haven't found anything that could help me.

Thanks!

**Comment by Machiventa** [comment]
How do I get 256kbps? Mine is downloading at 128

**Reply by patperek** [reply]
I believe if you have a Youtube Premium account it gives you the option to download at the highest quality available on the service, which is 256kbps m4a AAC. This can either be saved as the source file, or converted by the built in ffmpeg in the app.

**Reply by Machiventa** [reply]
Hmmm I have premium and see no option to dl high quality, and no option to increase to 256

**Reply by patperek** [reply]
Strange - I never changed any options, I've always had the option to download as 256kbps, and I've also always used the application with premium accounts only.

Not sure why it could be missing for you, sorry.

<img width="382" alt="Screenshot 2024-04-17 at 03 24 06" src="https://github.com/th-ch/youtube-music/assets/77973304/6594ba31-8081-4be8-9f18-c69a8ed0fd5d">

**Comment by mecki77** [comment]
> Hello, I've noticed that currently the downloader plugin will write album covers to downloaded mp3 files (mp3 256kbps), but if you do any other file format including "Source" file format (m4a), no album covers are applied to the files.
> 
> Is there any solution/config option to enable this? I've looked around on relevant discussions but haven't found anything that could help me.
> 
> Thanks!

MP3 is converted by the Download Add-on from the Source M4A-File and while it converts the file in MP3 it writes the Album cover, which is downloaded from youtube into the M4A-File. The Album Art is not included into the m4a Files - if you download them with jdownloader they have no cover too. So if you download them as source, the add-on downloads them "like they are" without cover and without adding the cover afterwards on your PC. You have to edit the files after you have them downloaded. But you can use the Windows Program mp3tag (https://www.mp3tag.de/) to edit the Tags in m4a Files and add Album art like Album Covers too. And you don't need the Cover picture to add it - mp3tag has several options to find it online inside the mp3tag Program. And at least you can add the covers for all files of an album in one step. 
m4a is the standard file format for youtube music - m4a is better then mp3 but unfortunately it is still less compatible then mp3.
I hope this helps.

**Reply by patperek** [reply]
Hi, yeah thanks for the help! I took a similar approach to what you suggested, and I'm now using an external app to add all the album covers and missing metadata, it works pretty well.
